app/views/layouts/_header.html.erb

```diff
@@ -112,19 +112,25 @@
   <div class="modal-dialog modal-lg modal-dialog-scrollable">
     <div class="modal-content">
       <div class="modal-header">
-        <h1 class="modal-title fs-5" id="select_language_dialog_label"><%= t ".select_language" %></h1>
+        <h1 class="modal-title fs-5" id="select_language_dialog_label"><%= t "languages_panes.select_language_list.header.select_language" %></h1>
         <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="<%= t "javascripts.close" %>"></button>
       </div>
       <div class="modal-body px-1">
-        <% if current_user&.id %>
-          <%= form_tag basic_preferences_path, :method => "PUT" do %>
-            <%= hidden_field_tag "referer", request.fullpath %>
-            <%= hidden_field_tag "language", I18n.locale %>
-            <%= render "layouts/select_language_list" %>
-          <% end %>
-        <% else %>
-          <%= render "layouts/select_language_list" %>
-        <% end %>
+        <div class="mb-3 px-2">
+          <%= text_field_tag :language_search,
+                             params[:language_search],
+                             :placeholder => t("languages_panes.select_language_list.search_language", :default => "Search..."),
+                             :autocomplete => "off",
+                             :dir => "auto",
+                             :class => "form-control py-2 px-3" %>
+        </div>
+        <turbo-frame id="select_language_list" src="<%= languages_pane_path %>?source=<%= ERB::Util.url_encode(request.fullpath) %>" loading="lazy">
+          <div class="text-center py-3">
+            <div class="spinner-border text-secondary" role="status">
+              <span class="visually-hidden">Loading...</span>
```

Comment:
Presumably `Loading...` here should be translatable...

Reply:
We have something similar in app/views/browse/_relation.html.erb. I think the text would be relevant for screen readers only, and it not visible otherwise.

<img width="852" height="138" alt="Image" src="https://github.com/user-attachments/assets/6bbc7e65-1d59-4524-a166-b297337df7db" />


<img width="829" height="265" alt="Image" src="https://github.com/user-attachments/assets/32bcaa22-8184-489b-a17a-a17b5dd07026" />

This pattern is following the example here: https://getbootstrap.com/docs/5.0/components/spinners/#border-spinner